config/src/main/scala/io/buoyant/config/Config.scala

```diff
@@ -0,0 +1,9 @@
+package io.buoyant.config
+
+import com.fasterxml.jackson.annotation.{JsonProperty, JsonTypeInfo}
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.EXISTING_PROPERTY, property = "kind", visible = true)
```

Comment:
Whoa!  Are annotations inherited?  If this works, that's pretty awesome.

There are many more config classes (e.g. RouterConfig and many others) and we should probably be consistent about having them all inherit this.